### ISSUE TITLE
windows not working

### ISSUE THREAD

**Issue by Goddard** [brandly/irc#72]
npm start

> irc@0.5.0 start E:\Source\irc
> npm run sass && NODE_ENV=production electron ./

> irc@0.5.0 sass E:\Source\irc
> node-sass src/sass/app.scss --output src/static/

Rendering Complete, saving .css file...
Wrote CSS to E:\Source\irc\src\static\app.css
'NODE_ENV' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! irc@0.5.0 start: `npm run sass && NODE_ENV=production electron ./`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the irc@0.5.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:

**Comment by brandly** [comment]
hey! thanks for trying to run this.

i've only ran this on a mac, and i don't know much about windows. on the line that's failing, i'm trying to set the `NODE_ENV` environment variable.

similarly, `npm run build` would probably fail on this line: https://github.com/brandly/irc/blob/38725e4eff696393911c867aab09abcdcd99d191/bin/build#L23

i guess i could be using [something like this](https://github.com/kentcdodds/cross-env).

in the meantime, you could remove `NODE_ENV=production` from that line, which will cause things to run _slightly_ slower, but maybe it'll work for you. let me know!

**Comment by brandly** [comment]
any update on this?

**Comment by Goddard** [comment]
Not running the same system any longer.  I am on Linux now.

**Comment by brandly** [comment]
Ah okay. I’ll leave this open in case someone else wants to try. Have a
good one!

**Comment by blckfalcon** [comment]
> i guess i could be using [something like this](https://github.com/kentcdodds/cross-env).

As for `npm start` using `cross-env` works. Also note that [index.js#L14](https://github.com/brandly/Lax/blob/master/index.js#L14) and changing into to some check like `app.dock.setIcon` for macOS and `win.setIcon` for Win/Linux I was able to run the app.

For `npm build` the build script would not work on Windows( not considering WSL ). Looking into it seems like doing much as job of a bundler(like webpack/parcel/etc), have you considered using it?

**Comment by brandly** [comment]
Thanks for chiming in! You have it running on Windows? I’d love to see a branch if you’ve got it. I just don’t know much about other platforms. 

It’s true, the build script does a lot of bundling. It doesn’t actually “bundle” the JS, as in it compiles each individual file with Babel but leaves them as separate files. 

The current setup grew one step at a time. Currently it supports a nice dev environment, as well as smooth builds that are released with GitHub Actions. Know any projects doing this with cross-platform support?

**Comment by blckfalcon** [comment]
Yeah I got it running just to give a quick look. I've made changes locally, it's nothing much regarding the start, I can put on a fork for you to see it, if you want I could also make as a PR.

And no, I don't know much of any other projects like this. Just made a search on GitHub involving irc and electron and your project showed up and decided to look around. Cool project you have btw.

**Comment by brandly** [comment]
Thanks!! It’s gone thru a few iterations over the years and grew into something nice. I guess Redux has kinda fallen out of fashion, but it’s working great here. 

I think a lot of IRC folks consider electron to be bloated and unnecessary, but most IRC clients aren’t very welcoming!! Just wanted to make something friendly. 

I originally wanted this to be cross-platform, but I’m always on a Mac so it’s tough. I’m under the impression that Windows now like has a Linux kernel and supports bash? I was hoping electron would hide platform differences from me lol

Maybe the best thing I can do is write out more details about how the macos build works. Someone like you could then write a separate build script for Windows? We’re trying to produce an .exe?

Idk what’s best but let me know what you think

**Comment by blckfalcon** [comment]
> I’m under the impression that Windows now like has a Linux kernel and supports bash? I was hoping electron would hide platform differences from me lol

I think WSL2 will support linux kernel inside windows, but that's still in development for now. Windows has also git-bash, which gives some bash support, but it does gives a different error when running your build script:
```
installing production dependencies into build/
npm ERR! code ENOSELF
npm ERR! Refusing to install package with name "Lax" under a package
npm ERR! also called "Lax". Did you name your project the same
npm ERR! as the dependency you're installing?
```
Since I am not well familiarized with the way it builds or how npm does stuff, I don't know how to help you here.

> Idk what’s best but let me know what you think

IMHO let that build process/task be handled by javascript side if possible.

**Comment by blckfalcon** [comment]
[Here](https://github.com/blckfalcon/Lax/commits/fr-build-gulp-webpack), in this branch, I've made some changes to the build process to be handled via gulp+webpack and it produces the windows binaries, it should do for mac and linux too, although I didn't test in these platforms. It would be cool if it works for you right out the box, if not shouldn't be much of a change.

It may not be your ideal solution for this, but if you want it I could make a PR.

**Comment by brandly** [comment]
it looks like i can [install wine and build a windows app on macos](https://github.com/electron/electron-packager#building-windows-apps-from-non-windows-platforms). i modified `bin/build` to specify `--platform win32` and no `--icon`, and it produced various files including a `.exe`. i zipped 



cc @MitchTalmadge

**Comment by MitchTalmadge** [comment]
It works @brandly! Good work :)

**Comment by brandly** [comment]
good to hear. i didn't specify an `--icon` to `electron-packager`, and i'm not sure how to build this in a github workflow yet. i'll probably update the readme to indicate there's an available `.exe`, but i'll leave this issue open since it's not fully solved.
